core/backend/src/ElementTreeWalker.ts

```diff
@@ -450,17 +450,22 @@ export class ElementSubTreeDeleter extends ElementTreeTopDown {
 /** Deletes an element tree starting with the specified top element. The top element is also deleted. Uses ElementTreeDeleter.
  * @param iModel The iModel
  * @param topElement The parent of the sub-tree
+ * @param maxPasses Maximum number of passes to make when deleting definitions
  * @beta
  */
-export function deleteElementTree(iModel: IModelDb, topElement: Id64String): void {
-  const del = new ElementTreeDeleter(iModel);
-  del.deleteNormalElements(topElement);
-  del.deleteSpecialElements();
+export function deleteElementTree(iModel: IModelDb, topElement: Id64String, maxPasses: number = 5): void {
+  let pass = 0;
+  do {
+    const del = new ElementTreeDeleter(iModel);
+    del.deleteNormalElements(topElement);
```

Comment:
Can you explain this to me? The comment above deleteElementTree mentions the top element is also deleted, yet you now have to try and delete the element many times, because it can't be deleted until all of its definitionElements have been deleted? And deleting some definitionElements may make other definitionElements unused, allowing you to delete more?

Reply:
Yes, that's right. There may be definitions in DefinitionPartitions/Containers that are children of topElement. If those definitions cannot be deleted because they are still in use, then the parent, topElement, can't be deleted. So, checking to see if topElement was not deleted is the indirect way of telling that some of its child definitions could not be deleted.